src/components/constraints/ConstraintForm.svelte

```diff
@@ -45,19 +52,30 @@
     plan_id: constraintPlanId,
   };
 
-  $: saveButtonEnabled =
-    constraintDefinition !== '' && constraintName !== '' && (constraintModelId !== null || constraintPlanId !== null);
   $: constraintModified = diffConstraints(savedConstraint, {
     definition: constraintDefinition,
     description: constraintDescription,
     model_id: constraintModelId,
     name: constraintName,
     plan_id: constraintPlanId,
   });
-  $: saveButtonText = mode === 'edit' && !constraintModified ? 'Saved' : 'Save';
-  $: saveButtonClass = saveButtonEnabled && constraintModified ? 'primary' : 'secondary';
+  $: {
+    if (constraintPlanId !== null) {
+      hasPermission = hasPlanPermission(initialPlanMap[constraintPlanId], mode, user);
+    } else if (constraintModelId !== null) {
+      hasPermission = hasModelPermission(constraintModelId, mode, user);
+    } else {
+      hasPermission = plans.reduce((prevPermission: boolean, plan) => {
+        return prevPermission || hasPlanPermission(plan, mode, user);
+      }, false);
+    }
+  }
   $: pageTitle = mode === 'edit' ? 'Constraints' : 'New Constraint';
   $: pageSubtitle = mode === 'edit' ? savedConstraint.name : '';
+  $: saveButtonEnabled =
```

Comment:
Thoughts on disabling the save button entirely + adding a tooltip if the user doesn't have constraint editing permissions?

Reply:
Oh I missed that. Thanks! I'll add it

Reply:
@AaronPlave I updated the button to take the permissionHandler action. Thanks!